Agent names not properly synchronized across cluster nodes (appear as unknown in global.db and cluster_control)

|Wazuh version|Component|Install type|Install method|
|---|---|---|---|
| 4.13.0-beta1 | cluster | Manager/Agent | Packages/Sources |

### Description:

During the test for [Release 4.13.0 - Beta 1 - System tests](https://github.com/wazuh/wazuh/issues/30702), we encountered a synchronization issue where some fields from the agent table in the `global.db` database are not correctly updated or propagated across all cluster nodes. Specifically, the `name` field for certain agents is shown as `unknown` on the master and workers.

[all_logs_and_environment_configuration.zip](https://github.com/user-attachments/files/21172013/cluster_logs.zip)

```console
root@wazuh-master: ~# /var/ossec/bin/cluster_control -a
ID   NAME              IP          STATUS  VERSION        NODE NAME      
000  wazuh-master      127.0.0.1   active  Wazuh v4.13.0  wazuh-master   
034  unknown           172.18.0.5  active  Wazuh v4.13.0  wazuh-worker2  
035  unknown           172.18.0.6  active  Wazuh v4.13.0  wazuh-worker1  
036  Agent-1752172081  172.18.0.7  active  Wazuh v4.13.0  wazuh-worker1
```

```console
root@wazuh-master: ~# sqlite3 -header -column /var/ossec/queue/db/global.db "SELECT id, name, ip, node_name, [group] FROM agent;"
id          name          ip          node_name     group     
----------  ------------  ----------  ------------  ----------
0           wazuh-master  127.0.0.1   wazuh-master            
34          unknown       172.18.0.5  wazuh-worker  Group1    
35          unknown       172.18.0.6  wazuh-worker  Group2    
36          Agent-175217  172.18.0.7  wazuh-worker  Group3

root@wazuh-worker1: ~# sqlite3 -header -column /var/ossec/queue/db/global.db "SELECT id, name, ip, node_name, [group] FROM agent;"
id          name           ip          node_name      group     
----------  -------------  ----------  -------------  ----------
0           wazuh-worker1  127.0.0.1   wazuh-worker1            
34          unknown        0.0.0.0     unknown        Group1    
35          unknown        172.18.0.6  wazuh-worker1  Group2    
36          Agent-1752172  172.18.0.7  wazuh-worker1  Group3 

root@wazuh-worker2: ~# sqlite3 -header -column /var/ossec/queue/db/global.db "SELECT id, name, ip, node_name, [group] FROM agent;"
id          name           ip          node_name      group     
----------  -------------  ----------  -------------  ----------
0           wazuh-worker2  127.0.0.1   wazuh-worker2            
34          unknown        172.18.0.5  wazuh-worker2  Group1    
35          Agent-1752172              unknown        Group2    
36          Agent-1752172              unknown        Group3
```



### Environment:

- **Master node:** wazuh-master
- **Worker nodes:** wazuh-worker1, wazuh-worker2
- **Agents:**
  - agent1 (connected to worker2, added to Group1)
  - agent2 (connected to worker1, added to Group2)
  - agent3 (connected to worker1, added to Group3)


 ### Steps to reproduce

1. Configure a Wazuh cluster with 1 master and 2 workers.
2. Add three different agents, each assigned to a unique group (Group1, Group2, Group3).
3. Restart all agents after registration and group assignment.
4. Run the following command from the master node:
  `/var/ossec/bin/cluster_control -a`

### Actual Result:

Only one agent shows the correct name (Agent-1752172081), while the others appear as unknown:
```console
ID   NAME              IP          STATUS  VERSION        NODE NAME      
000  wazuh-master      127.0.0.1   active  Wazuh v4.13.0  wazuh-master   
034  unknown           172.18.0.5  active  Wazuh v4.13.0  wazuh-worker2  
035  unknown           172.18.0.6  active  Wazuh v4.13.0  wazuh-worker1  
036  Agent-1752172081  172.18.0.7  active  Wazuh v4.13.0  wazuh-worker1

```

### Expected Result:

All agents should appear in the cluster list with correct names and IP addresses:
```console
ID    NAME                    IP          STATUS  VERSION        NODE NAME
034  Agent-1752172081         172.18.0.5  active  Wazuh v4.13.0  wazuh-worker2
035  Agent-1752172081         172.18.0.6  active  Wazuh v4.13.0  wazuh-worker1
036  Agent-1752172081         172.18.0.7  active  Wazuh v4.13.0  wazuh-worker1
```


### Additional information:

```console
root@wazuh-master: ~# /var/ossec/bin/wazuh-control status
wazuh-clusterd is running...
wazuh-modulesd is running...
wazuh-monitord is running...
wazuh-logcollector is running...
wazuh-remoted is running...
wazuh-syscheckd is running...
wazuh-analysisd is running...
wazuh-maild not running...
wazuh-execd is running...
wazuh-db is running...
wazuh-authd is running...
wazuh-agentlessd not running...
wazuh-integratord not running...
wazuh-dbd not running...
wazuh-csyslogd not running...
wazuh-apid is running...

root@wazuh-master: ~# /var/ossec/bin/wazuh-control info  
WAZUH_VERSION="v4.13.0"
WAZUH_REVISION="beta1"
WAZUH_TYPE="server"

root@wazuh-master: ~# /var/ossec/bin/cluster_control -l
NAME           TYPE    VERSION  ADDRESS       
wazuh-master   master  4.13.0   wazuh-master  
wazuh-worker1  worker  4.13.0   172.18.0.3    
wazuh-worker2  worker  4.13.0   172.18.0.4  

root@wazuh-master: ~# /var/ossec/bin/cluster_control -i 
Cluster name: wazuh

Last completed synchronization for connected nodes (2):
    wazuh-worker1 (172.18.0.3): Integrity check: 2025-07-10T22:40:05.492990Z | Integrity sync: n/a | Agents-info: 2025-07-10T22:39:59.660754Z | Agent-groups: n/a | Agent-groups full: n/a | Last keep alive: 2025-07-10T22:40:01.474984Z.
    wazuh-worker2 (172.18.0.4): Integrity check: 2025-07-10T22:39:59.033133Z | Integrity sync: n/a | Agents-info: 2025-07-10T22:39:58.461586Z | Agent-groups: n/a | Agent-groups full: n/a | Last keep alive: 2025-07-10T22:39:14.965798Z.

root@wazuh-master: ~# /var/ossec/bin/agent_control -l

Wazuh agent_control. List of available agents:
   ID: 000, Name: wazuh-master (server), IP: 127.0.0.1, Active/Local
   ID: 034, Name: Agent-1752172069, IP: 172.18.0.5, Active
   ID: 035, Name: Agent-1752172075, IP: 172.18.0.6, Active
   ID: 036, Name: Agent-1752172081, IP: 172.18.0.7, Active

List of agentless devices:

root@wazuh-master: ~# /var/ossec/bin/agent_groups 
Groups (4):
  Group1 (1)
  Group2 (1)
  Group3 (1)
  default (0)
Unassigned agents: 0.

root@wazuh-master: ~# /var/ossec/bin/manage_agents 


****************************************
* Wazuh v4.13.0 Agent manager.         *
* The following options are available: *
****************************************
   (A)dd an agent (A).
   (E)xtract key for an agent (E).
   (L)ist already added agents (L).
   (R)emove an agent (R).
   (Q)uit.
Choose your action: A,E,L,R or Q: L

Available agents: 
   ID: 034, Name: Agent-1752172069, IP: 172.18.0.5
   ID: 035, Name: Agent-1752172075, IP: 172.18.0.6
   ID: 036, Name: Agent-1752172081, IP: 172.18.0.7

** Press ENTER to return to the main menu.

root@wazuh-master: ~# /var/ossec/bin/agent_control -i 034

Wazuh agent_control. Agent information:
   Agent ID:   034
   Agent Name: Agent-1752172069
   IP address: 172.18.0.5
   Status:     Active

   Operating system:    Linux |wazuh-agent1 |6.8.0-60-generic |#63~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 22 19:00:15 UTC 2 |x86_64
   Client version:      Wazuh v4.13.0
   Configuration hash:  ab73af41699f13fdd81903b5f23d8d00
   Shared file hash:    1343e494a0f538df2ae2365d08df354c
   Last keep alive:     1752187088

   Syscheck last started at:  Unknown
   Syscheck last ended at:    Unknown


root@wazuh-master: ~# /var/ossec/bin/agent_control -i 035

Wazuh agent_control. Agent information:
   Agent ID:   035
   Agent Name: Agent-1752172075
   IP address: 172.18.0.6
   Status:     Active

   Operating system:    Linux |wazuh-agent2 |6.8.0-60-generic |#63~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 22 19:00:15 UTC 2 |x86_64
   Client version:      Wazuh v4.13.0
   Configuration hash:  ab73af41699f13fdd81903b5f23d8d00
   Shared file hash:    2fc71c175b51f3086ba5b721a65768ee
   Last keep alive:     1752187109

   Syscheck last started at:  Unknown
   Syscheck last ended at:    Unknown

root@wazuh-master: ~# /var/ossec/bin/agent_control -i 036

Wazuh agent_control. Agent information:
   Agent ID:   036
   Agent Name: Agent-1752172081
   IP address: 172.18.0.7
   Status:     Active

   Operating system:    Linux |wazuh-agent3 |6.8.0-60-generic |#63~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 22 19:00:15 UTC 2 |x86_64
   Client version:      Wazuh v4.13.0
   Configuration hash:  ab73af41699f13fdd81903b5f23d8d00
   Shared file hash:    17badba575121c82baf8e2990d25d5a8
   Last keep alive:     1752187119

   Syscheck last started at:  Unknown
   Syscheck last ended at:    Unknown

```

I am reopening this issue because the same behavior has been observed in [RC 2](https://github.com/wazuh/wazuh/issues/30990) testing:

When we check the status of the agents, we get the following again:

```
E           AssertionError:  Did not recieve expected agent status 008  Agent-1753170680  172.18.0.5  active in data ID   NAME              IP          STATUS  VERSION        NODE NAME      
E           000  wazuh-master      127.0.0.1   active  Wazuh v4.13.0  wazuh-master   
E           008  unknown           172.18.0.5  active  Wazuh v4.13.0  wazuh-worker1  
E           009  Agent-1753170688  172.18.0.6  active  Wazuh v4.13.0  wazuh-worker1  
E           010  Agent-1753170696  172.18.0.7  active  Wazuh v4.13.0  wazuh-worker1
```


Evidences: [basic_cluster.zip](https://github.com/user-attachments/files/21370006/basic_cluster.zip)

--- a/src/unit_tests/wazuh_db/test_wdb_global.c
+++ b/src/unit_tests/wazuh_db/test_wdb_global.c
@@ -9109,6 +9109,7 @@ void test_wdb_global_validate_groups_success(void **state) {
 void test_wdb_global_set_agent_groups_override_success(void **state) {
     test_struct_t *data  = (test_struct_t *)*state;
     int agent_id = 1;
+    const char * agent_name = "agent001";
     int group_id = 1;
     char group_name[] = "GROUP";
     char hash[] = "19dcd0dd"; //"GROUP" hash
@@ -9123,6 +9124,7 @@ void test_wdb_global_set_agent_groups_override_success(void **state) {
     for (int i=0; i<AGENTS_SIZE; i++) {
         cJSON* j_agent_group = cJSON_CreateObject();
         cJSON_AddItemToObject(j_agent_group, "id", cJSON_CreateNumber(agent_id));
+        cJSON_AddItemToObject(j_agent_group, "name", cJSON_CreateString(agent_name));
         cJSON_AddItemToObject(j_agent_group, "groups", cJSON_Duplicate(j_group_array, TRUE));
         cJSON_AddItemToArray(j_agents_group_info, j_agent_group);
 
@@ -9159,6 +9161,7 @@ void test_wdb_global_set_agent_groups_override_success(void **state) {
 void test_wdb_global_set_agent_groups_override_delete_error(void **state) {
     test_struct_t *data  = (test_struct_t *)*state;
     int agent_id = 1;
+    const char * agent_name = "agent001";
     int group_id = 1;
     char group_name[] = "GROUP";
     char hash[] = "19dcd0dd"; //"GROUP" hash
@@ -9173,6 +9176,7 @@ void test_wdb_global_set_agent_groups_override_delete_error(void **state) {
     for (int i=0; i<AGENTS_SIZE; i++) {
         cJSON* j_agent_group = cJSON_CreateObject();
         cJSON_AddItemToObject(j_agent_group, "id", cJSON_CreateNumber(agent_id));
+        cJSON_AddItemToObject(j_agent_group, "name", cJSON_CreateString(agent_name));
         cJSON_AddItemToObject(j_agent_group, "groups", cJSON_Duplicate(j_group_array, TRUE));
         cJSON_AddItemToArray(j_agents_group_info, j_agent_group);
 
@@ -9212,6 +9216,7 @@ void test_wdb_global_set_agent_groups_override_delete_error(void **state) {
 void test_wdb_global_set_agent_groups_add_modes_assign_error(void **state) {
     test_struct_t *data  = (test_struct_t *)*state;
     int agent_id = 1;
+    const char * agent_name = "agent001";
     char group_name[] = "GROUP";
     char hash[] = "19dcd0dd"; //"GROUP" hash
     char sync_status[] = "synced";
@@ -9227,6 +9232,7 @@ void test_wdb_global_set_agent_groups_add_modes_assign_error(void **state) {
     for (int i=0; i<AGENTS_SIZE; i++) {
         cJSON* j_agent_group = cJSON_CreateObject();
         cJSON_AddItemToObject(j_agent_group, "id", cJSON_CreateNumber(agent_id));
+        cJSON_AddItemToObject(j_agent_group, "name", cJSON_CreateString(agent_name));
         cJSON_AddItemToObject(j_agent_group, "groups", cJSON_Duplicate(j_group_array_invalid, TRUE));
         cJSON_AddItemToArray(j_agents_group_info, j_agent_group);
 
@@ -9872,6 +9878,7 @@ void test_wdb_global_recalculate_all_agent_groups_hash_recalculate_error(void **
 void test_wdb_global_assign_agent_group_agent_not_exists_success(void **state) {
     test_struct_t *data  = (test_struct_t *)*state;
     int agent_id = 1;
+    const char* agent_name = "agent-01";
     int group_id = 1;
     int priority = 0;
     char group_name[] = "test_group";
@@ -9922,7 +9929,7 @@ void test_wdb_global_assign_agent_group_agent_not_exists_success(void **state) {
     expect_value(__wrap_sqlite3_bind_int, value, agent_id);
     will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
     expect_value(__wrap_sqlite3_bind_text, pos, 2); // name
-    expect_string(__wrap_sqlite3_bind_text, buffer, "unknown");
+    expect_string(__wrap_sqlite3_bind_text, buffer, agent_name);
     will_return(__wrap_sqlite3_bind_text, SQLITE_OK);
     expect_value(__wrap_sqlite3_bind_text, pos, 3); // ip
     expect_string(__wrap_sqlite3_bind_text, buffer, "0.0.0.0");
@@ -9953,7 +9960,7 @@ void test_wdb_global_assign_agent_group_agent_not_exists_success(void **state) {
     will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
     will_return(__wrap_wdb_exec_stmt_silent, OS_SUCCESS);
 
-    wdbc_result result = wdb_global_assign_agent_group(data->wdb, agent_id, j_groups, priority, true);
+    wdbc_result result = wdb_global_assign_agent_group(data->wdb, agent_id, j_groups, priority, agent_name);
 
     assert_int_equal(result, WDBC_OK);
     __real_cJSON_Delete(j_groups);
@@ -9962,6 +9969,7 @@ void test_wdb_global_assign_agent_group_agent_not_exists_success(void **state) {
 void test_wdb_global_assign_agent_group_agent_not_exists_insert_fail(void **state) {
     test_struct_t *data  = (test_struct_t *)*state;
     int agent_id = 1;
+    const char* agent_name = "agent-01";
     int group_id = 1;
     int priority = 0;
     char group_name[] = "test_group";
@@ -10012,7 +10020,7 @@ void test_wdb_global_assign_agent_group_agent_not_exists_insert_fail(void **stat
     expect_value(__wrap_sqlite3_bind_int, value, agent_id);
     will_return(__wrap_sqlite3_bind_int, SQLITE_OK);
     expect_value(__wrap_sqlite3_bind_text, pos, 2); // name
-    expect_string(__wrap_sqlite3_bind_text, buffer, "unknown");
+    expect_string(__wrap_sqlite3_bind_text, buffer, agent_name);
     will_return(__wrap_sqlite3_bind_text, SQLITE_OK);
     expect_value(__wrap_sqlite3_bind_text, pos, 3); // ip
     expect_string(__wrap_sqlite3_bind_text, buffer, "0.0.0.0");
@@ -10045,7 +10053,7 @@ void test_wdb_global_assign_agent_group_agent_not_exists_insert_fail(void **stat
 
     expect_string(__wrap__mdebug1, formatted_msg, "Unable to insert group 'test_group' for agent '1', retry failed after creating agent in never_connected state.");
 
-    wdbc_result result = wdb_global_assign_agent_group(data->wdb, agent_id, j_groups, priority, true);
+    wdbc_result result = wdb_global_assign_agent_group(data->wdb, agent_id, j_groups, priority, agent_name);
 
     assert_int_equal(result, WDBC_ERROR);
     __real_cJSON_Delete(j_groups);

--- a/src/wazuh_db/main.c
+++ b/src/wazuh_db/main.c
@@ -105,6 +105,8 @@ int main(int argc, char ** argv)
     wconfig.max_fragmentation = getDefine_Int("wazuh_db", "max_fragmentation", 0, 100);
     wconfig.check_fragmentation_interval = getDefine_Int("wazuh_db", "check_fragmentation_interval", 1, 30758400);
 
+    wconfig.is_worker_node = w_is_worker() == 1;
+
     // Allocating memory for configuration structures and setting default values
     wdb_init_conf();
 

--- a/src/wazuh_db/wdb.c
+++ b/src/wazuh_db/wdb.c
@@ -191,7 +191,7 @@ static const char *SQL_STMT[] = {
     [WDB_STMT_GLOBAL_SYNC_REQ_KEEPALIVE_GET] = "SELECT id, last_keepalive FROM agent WHERE id > ? AND sync_status = 'syncreq_keepalive' LIMIT 1;",
     [WDB_STMT_GLOBAL_SYNC_GET] = "SELECT sync_status FROM agent WHERE id = ?;",
     [WDB_STMT_GLOBAL_SYNC_SET] = "UPDATE agent SET sync_status = ? WHERE id = ?;",
-    [WDB_STMT_GLOBAL_GROUP_SYNC_REQ_GET] = "SELECT id FROM agent WHERE id > ? AND group_sync_status = 'syncreq' AND date_add < ? LIMIT 1;",
+    [WDB_STMT_GLOBAL_GROUP_SYNC_REQ_GET] = "SELECT id, name FROM agent WHERE id > ? AND group_sync_status = 'syncreq' AND date_add < ? LIMIT 1;",
     [WDB_STMT_GLOBAL_GROUP_SYNC_ALL_GET] = "SELECT id FROM agent WHERE id > ? AND date_add < ? LIMIT 1;",
     [WDB_STMT_GLOBAL_GROUP_SYNCREQ_FIND] = "SELECT 1 FROM agent WHERE group_sync_status = 'syncreq';",
     [WDB_STMT_GLOBAL_AGENT_GROUPS_NUMBER_GET] = "SELECT count(id_group) groups_number from belongs WHERE id_agent = ?;",

--- a/src/wazuh_db/wdb.h
+++ b/src/wazuh_db/wdb.h
@@ -387,6 +387,7 @@ typedef struct wdb_config {
     int max_fragmentation;
     int check_fragmentation_interval;
     wdb_backup_settings_node** wdb_backup_settings;
+    bool is_worker_node; ///< Indicates if the node is a cluster worker node
 } wdb_config;
 
 /// Enumeration of components supported by the integrity library.
@@ -2190,10 +2191,11 @@ int wdb_global_get_agent_max_group_priority(wdb_t *wdb, int id);
  * @param [in] id ID of the agent to add new groups.
  * @param [in] j_groups JSON array with all the groups of the agent.
  * @param [in] priority Initial priority to insert the groups.
- * @param [in] add_agent If true and the agent doesn't exist, it will be created.
+ * @param [in] create_agent_name If not null and the agent doesn't exist, it will be created with the given name.
  * @return wdbc_result representing the status of the command.
  */
-wdbc_result wdb_global_assign_agent_group(wdb_t *wdb, int id, cJSON* j_groups, int priority, bool add_agent);
+wdbc_result
+wdb_global_assign_agent_group(wdb_t* wdb, int id, cJSON* j_groups, int priority, const char* create_agent_name);
 
 /**
  * @brief Deletes groups of an agent.

--- a/src/wazuh_db/wdb_global.c
+++ b/src/wazuh_db/wdb_global.c
@@ -1283,7 +1283,7 @@ int wdb_global_get_agent_max_group_priority(wdb_t *wdb, int id) {
     return group_priority;
 }
 
-wdbc_result wdb_global_assign_agent_group(wdb_t *wdb, int id, cJSON* j_groups, int priority, bool add_agent) {
+wdbc_result wdb_global_assign_agent_group(wdb_t *wdb, int id, cJSON* j_groups, int priority, const char* create_agent_name) {
     cJSON* j_group_name = NULL;
     wdbc_result result = WDBC_OK;
     cJSON_ArrayForEach (j_group_name, j_groups) {
@@ -1294,7 +1294,7 @@ wdbc_result wdb_global_assign_agent_group(wdb_t *wdb, int id, cJSON* j_groups, i
                 cJSON* j_group_id = cJSON_GetObjectItem(j_find_response->child, "id");
                 if (cJSON_IsNumber(j_group_id)) {
                     int insert_result = wdb_global_insert_agent_belong(wdb, j_group_id->valueint, id, priority);
-                    if (!add_agent && OS_INVALID == insert_result) {
+                    if (!create_agent_name && OS_INVALID == insert_result) {
                         // If the agent doesn't exist, we don't want to insert the group relationship.
                         insert_result = OS_NOTFOUND;
                     }
@@ -1302,7 +1302,7 @@ wdbc_result wdb_global_assign_agent_group(wdb_t *wdb, int id, cJSON* j_groups, i
                         // Check if agent exists
                         if (!wdb_global_agent_exists(wdb, id)) {
                             // Create agent in never_connected state
-                            const char *unknown = "unknown";
+                            const char *unknown = create_agent_name ? create_agent_name : "unknown";
                             const char *ip = "0.0.0.0";
                             if (OS_INVALID == wdb_global_insert_agent(wdb, id, (char*)unknown, (char*)ip, (char*)ip, NULL, NULL, time(NULL))) {
                                 mdebug1("Unable to create agent '%d' in never_connected state", id);
@@ -1388,7 +1388,7 @@ int wdb_global_if_empty_set_default_agent_group(wdb_t *wdb, int id) {
     if (OS_INVALID == wdb_global_get_agent_max_group_priority(wdb, id)) {
         cJSON* j_default_group = cJSON_CreateArray();
         cJSON_AddItemToArray(j_default_group, cJSON_CreateString("default"));
-        if (WDBC_OK == wdb_global_assign_agent_group(wdb, id, j_default_group, 0, false)) {
+        if (WDBC_OK == wdb_global_assign_agent_group(wdb, id, j_default_group, 0, NULL)) {
             mdebug1("Agent '%03d' reassigned to 'default' group", id);
         } else {
             merror("There was an error assigning the agent '%03d' to default group", id);
@@ -1513,7 +1513,25 @@ wdbc_result wdb_global_set_agent_groups(wdb_t *wdb, wdb_groups_set_mode_t mode, 
                     }
                 }
                 if (valid_groups = wdb_global_validate_groups(wdb, j_groups, agent_id), OS_SUCCESS == valid_groups) {
-                    if (WDBC_ERROR == wdb_global_assign_agent_group(wdb, agent_id, j_groups, group_priority, mode == WDB_GROUP_OVERRIDE ? true : false)) {
+                    char* agent_name = NULL;
+
+                    // For override mode, require agent name
+                    if (mode == WDB_GROUP_OVERRIDE) {
+                        cJSON* j_agent_name = cJSON_GetObjectItem(j_group_info, "name");
+                        if (j_agent_name && cJSON_IsString(j_agent_name)) {
+                            agent_name = j_agent_name->valuestring;
+                        } else if (wconfig.is_worker_node) {
+                            merror("Agent name is required when overriding groups for agent '%03d'", agent_id);
+                            ret = WDBC_ERROR;
+                            continue;
+                        } else {
+                            mdebug1("Agent name is not provided for agent '%03d', the agent not will not be created if "
+                                    "it does not exist",
+                                    agent_id);
+                        }
+                    }
+
+                    if (WDBC_ERROR == wdb_global_assign_agent_group(wdb, agent_id, j_groups, group_priority, agent_name)) {
                         ret = WDBC_ERROR;
                         merror("There was an error assigning the groups to agent '%03d'", agent_id);
                     }

--- a/tests/integration/test_wazuh_db/test_groups/data/test_cases/cases_sync_agent_groups_get.yaml
+++ b/tests/integration/test_wazuh_db/test_groups/data/test_cases/cases_sync_agent_groups_get.yaml
@@ -4,7 +4,7 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"condition":"sync_status"}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']}, {'id': 2, 'groups': ['Test_group2']}]}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']}, {'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}]}]"
 
 - name: Test sync_status without response
   description:
@@ -42,7 +42,7 @@
     pre_input:
       - global sql UPDATE agent SET group_sync_status="synced" WHERE id = 2
     input: global sync-agent-groups-get {"condition":"sync_status"}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']}]}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']}]}]"
 
 - name: Test 'all' condition when one agent groups are in 'synced'
   description:
@@ -76,7 +76,7 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"condition":"sync_status", "set_synced":true}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']}, {'id': 2, 'groups': ['Test_group2']}]}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']}, {'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}]}]"
     new_status: synced
     agent_id: "[1,2]"
 
@@ -86,7 +86,7 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"condition":"sync_status", "set_synced":false}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']}, {'id': 2, 'groups': ['Test_group2']}]}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']}, {'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}]}]"
     new_status: syncreq
     agent_id: "[1,2]"
 
@@ -106,8 +106,8 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"last_id":0, "condition":"sync_status", "get_global_hash":true}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']},
-             {'id': 2, 'groups': ['Test_group2']}], 'hash': '[GLOBAL_HASH]'}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']},
+             {'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}], 'hash': '[GLOBAL_HASH]'}]"
 
 - name: Test only get_global_hash
   description:
@@ -123,7 +123,7 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"last_id":0, "condition":"sync_status", "get_global_hash":false}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']}, {'id': 2, 'groups': ['Test_group2']}]}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']}, {'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}]}]"
 
 - name: Test get_global_hash with invalid value
   description:
@@ -139,7 +139,7 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"condition":"sync_status", "agent_registration_delta":0}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']}, {'id': 2, 'groups': ['Test_group2']}]}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']}, {'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}]}]"
 
 - name: Test 'agent_registration_delta' in 0 and all condition
   description:
@@ -158,7 +158,7 @@
       - global insert-agent {"id":5,"name":"Agent-test5","date_add":1545753642}
       - global set-agent-groups {"mode":"append","sync_status":"syncreq","data":[{"id":5,"groups":["Test_group5"]}]}
     input: global sync-agent-groups-get {"condition":"sync_status", "agent_registration_delta":10000}
-    output: "[{'data': [{'id': 5, 'groups': ['Test_group5']}]}]"
+    output: "[{'data': [{'id': 5, 'name': 'Agent-test5', 'groups': ['Test_group5']}]}]"
 
 - name: Test 'agent_registration_delta' with delta in 10000 and all
   description:
@@ -177,7 +177,7 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"last_id":0, "condition":"sync_status"}
-    output: "[{'data': [{'id': 1, 'groups': ['Test_group1']}, {'id': 2, 'groups': ['Test_group2']}]}]"
+    output: "[{'data': [{'id': 1, 'name': 'Agent-test1', 'groups': ['Test_group1']}, {'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}]}]"
 
 - name: Test last_id - obtain from second group
   description:
@@ -185,7 +185,7 @@
   metadata:
     pre_required_group: Test_group1,Test_group2
     input: global sync-agent-groups-get {"last_id":1, "condition":"sync_status"}
-    output: "[{'data': [{'id': 2, 'groups': ['Test_group2']}]}]"
+    output: "[{'data': [{'id': 2, 'name': 'Agent-test2', 'groups': ['Test_group2']}]}]"
 
 - name: Test last_id - with not exist id
   description:

